File: cpp/cluster_management/src/CreateSingleRegion.cpp
#include "CreateSingleRegion.h"

using namespace Aws;
using namespace Aws::DSQL;
using namespace Aws::DSQL::Model;

Aws::String createCluster(Aws::DSQL::DSQLClient& client, bool deletionProtectionEnabled, const std::map<Aws::String, Aws::String>& tags) {
    CreateClusterRequest request;
    request.SetDeletionProtectionEnabled(deletionProtectionEnabled);
    request.SetTags(tags);
    CreateClusterOutcome outcome = client.CreateCluster(request);

    const auto& clusterResult = outcome.GetResult().GetIdentifier();
    if (outcome.IsSuccess()) {
        std::cout << "Cluster Identifier: " << clusterResult << std::endl;
    } else {
        std::cerr << "Create operation failed: " << outcome.GetError().GetMessage() << std::endl;
    }
    return clusterResult;
}

File: go/cluster_management/create_single_region.go
package main

import (
	"context"

	dsql "github.com/aws/aws-sdk-go-v2/service/dsql"
)

func CreateCluster(client *dsql.Client, deleteProtection bool, tags map[string]string) (clusterStatus *dsql.CreateClusterOutput, err error) {

	input := dsql.CreateClusterInput{
		DeletionProtectionEnabled: &deleteProtection,
		Tags:                      tags,
	}

	clusterStatus, err = client.CreateCluster(context.Background(), &input)
	return
}
